staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go

```diff
@@ -0,0 +1,462 @@
+			--rule="foo.com/path*=svc:8080" \
+			--rule="bar.com/admin*=svc2:http"
+
+		# Create an ingress with TLS enabled using the default ingress certificate and different path types
+		kubectl create ingress ingtls --class=default \
+		   --rule="foo.com/=svc:https,tls" \
+		   --rule="foo.com/path/subpath*=othersvc:8080"
+		
+		# Create an ingress with TLS enabled using a specific secret and pathType as Prefix
+		kubectl create ingress ingsecret --class=default \
+		   --rule="foo.com/*=svc:8080,tls=secret1"
+		
+		# Create an ingress with a default backend
+		kubectl create ingress ingdefault --class=default \
+		   --default-backend=defaultsvc:http \
+		   --rule="foo.com/*=svc:8080,tls=secret1"
+
+		`))
+)
+
+// CreateIngressOptions is returned by NewCmdCreateIngress
+type CreateIngressOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+
+	PrintObj func(obj runtime.Object) error
+
+	Name             string
+	IngressClass     string
+	Rules            []string
+	Annotations      []string
+	DefaultBackend   string
+	Namespace        string
+	EnforceNamespace bool
+	CreateAnnotation bool
+
+	Client         networkingv1client.NetworkingV1Interface
+	DryRunStrategy cmdutil.DryRunStrategy
+	DryRunVerifier *resource.DryRunVerifier
+
+	FieldManager string
+
+	genericclioptions.IOStreams
+}
+
+// NewCreateIngressOptions creates the CreateIngressOptions to be used later
+func NewCreateIngressOptions(ioStreams genericclioptions.IOStreams) *CreateIngressOptions {
+	return &CreateIngressOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme),
+		IOStreams:  ioStreams,
+	}
+}
+
+// NewCmdCreateIngress is a macro command to create a new ingress.
+// This command is better known to users as `kubectl create ingress`.
+func NewCmdCreateIngress(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	o := NewCreateIngressOptions(ioStreams)
+
+	cmd := &cobra.Command{
+		Use:                   "ingress NAME --class=ingress-class --rule=host/path=service:port[,tls[=secret]] --rule=host/otherpath*=svc:namedport[,tls[=secret]] --default-backend=default-svc:default-port",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"ing"},
+		Short:                 ingressLong,
+		Long:                  ingressLong,
+		Example:               ingressExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, cmd, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run())
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddGeneratorFlags(cmd, "")
+	cmd.Flags().StringVar(&o.IngressClass, "class", o.IngressClass, "Ingress Class to be used")
+	cmd.Flags().StringArrayVar(&o.Rules, "rule", o.Rules, "Rule in format host/path=service:port[,tls=secretname]. Paths containing the leading character '*' are considered pathType=Prefix. tls argument is optional.")
```

Comment:
I think it's 'hard' because when the secret doesn't exists Ingress Controllers (don't know if all of them) fallback into the default secret.

Also this would be useless when using dry-run, so IMO create should try to create the object and rely on the apiserver validation. If the api-server does not return an error/warning when you create an YAML with a non-existing secret, it should not return an error/warning when you use kubectl create command either :)

Reply:
Also this was my first thought also, like 'should we validate the existence of a secret' but then I realized that for me, it doesn't make sense